drivers/overlay/encryption.go

```diff
@@ -224,7 +225,9 @@ func programSA(localIP, remoteIP net.IP, spi *spi, k *key, dir int, add bool) (f
 	if add {
 		action = "Adding"
 		xfrmProgram = ns.NlHandle().XfrmStateAdd
-		crypt = &netlink.XfrmStateAlgo{Name: "cbc(aes)", Key: k.value}
+		salt := make([]byte, 4)
+		binary.LittleEndian.PutUint32(salt, k.tag)
+		crypt = &netlink.XfrmStateAlgo{Name: "rfc4106(gcm(aes))", Key: append(k.value, salt...), ICVLen: 64}
```

Comment:
`For a given key, the IV MUST NOT repeat.  The most natural way to implement this is with a counter`

I assume we're not responsible for choosing the 8 octet IV? It is incredibly important to guarantee that for the lifetime of this key this IV does not repeat.